core/utility/string_validator.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities for validating string type."""
+
+
+class InvalidStringException(Exception):
+    """The specified string is invalid."""
+
+
+def is_ascii(string):
+    """Returns whether a string is in ASCII."""
+    return all(ord(c) < 128 for c in string)
+
+
+def invalid_string_reason(string):
+    """Determines the reason why a string is invalid. Accepts US-ASCII string or
+        a Unicode string.
+
+    Args:
+        string: str. The string to check.
+
+    Returns:
+        A string that indicates the reason why a string is invalid; otherwise
```

Comment:
str|None. A string that ...

Also, the second "a string" ---> "the input string"

Reply:
Done!